Tests/SwiftProtobufTests/Test_JSON.swift

```diff
@@ -299,6 +327,67 @@ class Test_JSON: XCTestCase, PBTestHelpers {
         assertJSONDecodeSucceeds("{\"singleInt64\":null}") {$0.singleInt64 == 0}
         assertJSONDecodeSucceeds("{\"singleInt64\":2147483648}") {$0.singleInt64 == 2147483648}
         assertJSONDecodeSucceeds("{\"singleInt64\":2147483648}") {$0.singleInt64 == 2147483648}
+
+        assertJSONDecodeFails("{\"singleInt64\":1")
+        assertJSONDecodeFails("{\"singleInt64\":\"")
+        assertJSONDecodeFails("{\"singleInt64\":\"1")
+        assertJSONDecodeFails("{\"singleInt64\":\"1\"")
+    }
+
+    func testSingleUInt64() {
+        assertJSONEncode("{\"singleUint64\":\"1\"}") {(o: inout MessageTestType) in
+            o.singleUint64 = 1
+        }
+        assertJSONEncode("{\"singleUint64\":\"4294967295\"}") {(o: inout MessageTestType) in
+            o.singleUint64 = UInt64(UInt32.max)
+        }
+        assertJSONEncode("{\"singleUint64\":\"18446744073709551615\"}") {(o: inout MessageTestType) in
+            o.singleUint64 = UInt64.max
+        }
+        // Parse unquoted 64-bit integers
+        assertJSONDecodeSucceeds("{\"singleUint64\":18446744073709551615}") {$0.singleUint64 == UInt64.max}
+        // Accept quoted 64-bit integers with backslash escapes in them
+        assertJSONDecodeSucceeds("{\"singleUint64\":\"184467\\u00344073709551615\"}") {$0.singleUint64 == UInt64.max}
+        // Reject unquoted 64-bit integers with backslash escapes
+        assertJSONDecodeFails("{\"singleUint64\":184467\\u00344073709551615}")
+        // Reject out-of-range integers, whether or not quoted
+        assertJSONDecodeFails("{\"singleUint64\":\"18446744073709551616\"}")
+        assertJSONDecodeFails("{\"singleUint64\":18446744073709551616}")
+        assertJSONDecodeFails("{\"singleUint64\":\"184467440737095516109\"}")
+        assertJSONDecodeFails("{\"singleUint64\":184467440737095516109}")
+
+        // Explicit 'null' is permitted, decodes to default
+        assertJSONDecodeSucceeds("{\"singleUint64\":null}") {$0.singleUint64 == 0}
+        // Quoted or unquoted numbers, positive or zero
+        assertJSONDecodeSucceeds("{\"singleUint64\":1}") {$0.singleUint64 == 1}
+        assertJSONDecodeSucceeds("{\"singleUint64\":\"1\"}") {$0.singleUint64 == 1}
+        assertJSONDecodeSucceeds("{\"singleUint64\":0}") {$0.singleUint64 == 0}
+        assertJSONDecodeSucceeds("{\"singleUint64\":\"0\"}") {$0.singleUint64 == 0}
+        // Protobuf JSON does not accept leading zeros
+        assertJSONDecodeFails("{\"singleUint64\":01}")
+        assertJSONDecodeFails("{\"singleUint64\":\"01\"}")
+        // But it does accept exponential (as long as result is integral)
+        assertJSONDecodeSucceeds("{\"singleUint64\":4.294967295e9}") {$0.singleUint64 == UInt32.max}
```

Comment:
@tbkka this doesn't compile since it is comparing an `UInt64` with a `UInt32`.  Did you want to use `UInt64(UInt32.max)` or to update the test to be the 64bit value?